src/renderer/tvgShape.h

```diff
@@ -77,7 +77,23 @@ struct Shape::Impl
         const Paint* target;
         auto method = shape->composite(&target);
         if (!target || method == tvg::CompositeMethod::ClipPath) return false;
-        if (target->pImpl->opacity == 255 || target->pImpl->opacity == 0) return false;
+        if (target->pImpl->opacity == 255 || target->pImpl->opacity == 0) {
+            if (target->identifier() == TVG_CLASS_ID_SHAPE) {
+                auto shape = static_cast<const Shape*>(target);
+                if (!shape->fill()) {
+                    uint8_t r, g, b, a;
+                    shape->fillColor(&r, &g, &b, &a);
+                    if (method == tvg::CompositeMethod::AlphaMask || method == tvg::CompositeMethod::InvAlphaMask) {
```

Comment:
1. Let's remove the namespace "tvg" for compactness.
2. Please check this logic:
```
if (a == 0 || a == 255) {
   if (method == CompositeMethod::LumaMask || method == CompositeMethod::InvLumaMask) {
      if (((r == 255 && g == 255 && b == 255) || (r == 0 && g == 0 && b == 0)) return false;
   } else return false;
}
```

Reply:
done